cloud/misc/virt_net.py

```diff
@@ -0,0 +1,543 @@
+
+    def get_net(self, entryid):
+        return self.conn.find_entry(entryid)
+
+    def list_nets(self, state=None):
+        results = []
+        for entry in self.conn.find_entry(-1):
+            if state:
+                if state == self.conn.get_status2(entry):
+                    results.append(entry.name())
+            else:
+                results.append(entry.name())
+        return results
+
+    def state(self):
+        results = []
+        for entry in self.list_nets():
+            state_blurb = self.conn.get_status(entry)
+            results.append("%s %s" % (entry,state_blurb))
+        return results
+
+    def autostart(self, entryid):
+        return self.conn.set_autostart(entryid, True)
+
+    def get_autostart(self, entryid):
+        return self.conn.get_autostart2(entryid)
+
+    def set_autostart(self, entryid, state):
+        return self.conn.set_autostart(entryid, state)
+
+    def create(self, entryid):
+        return self.conn.create(entryid)
+
+    def start(self, entryid):
+        return self.conn.create(entryid)
+
+    def stop(self, entryid):
+        return self.conn.destroy(entryid)
+
+    def destroy(self, entryid):
+        return self.conn.destroy(entryid)
+
+    def undefine(self, entryid):
+        return self.conn.undefine(entryid)
+
+    def status(self, entryid):
+        return self.conn.get_status(entryid)
+
+    def get_xml(self, entryid):
+        return self.conn.get_xml(entryid)
+
+    def define(self, entryid, xml):
+        return self.conn.define_from_xml(entryid, xml)
+
+    def info(self):
+        return self.facts(facts_mode='info')
+
+    def facts(self, facts_mode='facts'):
+        results = dict()
+        for entry in self.list_nets():
+            results[entry] = dict()
+            results[entry]["autostart"] = self.conn.get_autostart(entry)
+            results[entry]["persistent"] = self.conn.get_persistent(entry)
+            results[entry]["state"] = self.conn.get_status(entry)
+            results[entry]["bridge"] = self.conn.get_bridge(entry)
+            results[entry]["uuid"] = self.conn.get_uuid(entry)
+
+            try:
+                results[entry]["forward_mode"] = self.conn.get_forward(entry)
+            except ValueError as e:
+                pass
+
+            try:
+                results[entry]["domain"] = self.conn.get_domain(entry)
+            except ValueError as e:
+                pass
+
+            try:
+                results[entry]["macaddress"] = self.conn.get_macaddress(entry)
+            except ValueError as e:
+                pass
+
+        facts = dict()
+        if facts_mode == 'facts':
+            facts["ansible_facts"] = dict()
+            facts["ansible_facts"]["ansible_libvirt_networks"] = results
+        elif facts_mode == 'info':
+            facts['networks'] = results
+        return facts
+
+
+def core(module):
+
+    state     = module.params.get('state', None)
```

Comment:
May I give a little note here, I know explicit is better than implicit, but get() does set `None` as default, no need to set it explicitly.